### PR TITLE
HUD code refactoring

### PR DESCRIPTION
**Pull request by tmp64**
Valve put all HUD-related headers into hud.h that is included almost everywhere and any slight change to any HUD element class will cause the whole client library to rebuild. That takes quite some time.

I split up hud.h into smaller CHudX.h files, added forward declarations to them in `hud.h` and replaced HUD element variables with smart pointers.

I also renamed cpp files to match the classname and header file name.

### CODE REVIEW

File: cl_dll/msvc/client.vcxproj [hunk 1]
```diff
@@ -14,19 +14,19 @@
...
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v140_xp</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
     <CharacterSet>MultiByte</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>false</UseDebugLibraries>
-    <PlatformToolset>v140_xp</PlatformToolset>
+    <PlatformToolset>v141</PlatformToolset>
```
Comment (LevShisterov) [inline cl_dll/msvc/client.vcxproj]: Should be _xp to be usable in WinXP. You can just revert this change from PR.

File: cl_dll/hud_redraw.cpp [hunk 2]
```diff
@@ -79,12 +76,6 @@ void CHud::Think(void)
 		m_flMouseSensitivity = sensitivity->value * ((float)newfov / (float)default_fov->value) * CVAR_GET_FLOAT("zoom_sensitivity_ratio");
 	}
 
-	// think about default fov
-	if ( m_iFOV == 0 )
-	{  // only let players adjust up in fov,  and only if they are not overriden by something else
-		m_iFOV = max( default_fov->value, 90 );  
-	}
-
```
Comment (LevShisterov) [inline cl_dll/hud_redraw.cpp:87]: Intended change?

**Review by LevShisterov — CHANGES_REQUESTED** [review]
Hi! A lot of work, thanks!
Currently build server is offline. Have to set it up first.

File: cl_dll/hud_redraw.cpp (hunk 2 above)
Comment (tmp64) [reply cl_dll/hud_redraw.cpp:87]: Yeah. It's supposed to allow you only increase the FOV but the condition is true only if default_fov is 0. So I just removed it.

File: cl_dll/msvc/client.vcxproj (hunk 1 above)
Comment (tmp64) [reply cl_dll/msvc/client.vcxproj]: Ok, I will once MSVS finishes installing.

Comment (tmp64) [reply cl_dll/msvc/client.vcxproj]: Fixed.

**Comment by tmp64** [comment]
I was working with this code structure for almost two years and I came to a conclusion that it isn't very good. While I think it's better than what used to be as every HUD element has its own file, all these CHud*.h/.cpp files in the project root directory make it uncomfortable to browse and look for specific files.

I don't think it should be merged into master.

**Pull request closed without merging**